internal/pkg/manifest/svc.go

```diff
@@ -194,7 +196,7 @@ func (a *AdvancedCount) IgnoreRange() bool {
 
 func (a *AdvancedCount) hasAutoscaling() bool {
 	return !a.Range.IsEmpty() || a.CPU != nil || a.Memory != nil ||
-		a.Requests != nil || a.ResponseTime != nil
+		a.Requests != nil || a.ResponseTime != nil || !a.QueueScaling.IsEmpty()
```

Comment:
What if a non-worker service workload falsely specified `queue_delay`? Should we consider it as having autoscaling?

Reply:
ugh yea this is kindof an issue even today for the `requests` or `response_time` fields, what if a backend service only gives:
```yaml
count:
  range: 1-10
  requests: 1000
```
Then autoscaling resources will be created except the `ScalingPolicy` :| which might be a bit surprising to the users.

Let me see if I can think of a way of erroring out early 💭

Reply:
We could possibly do it in manifest validation to have this logic [here](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/validate.go#L19) 💭

Reply:
Or update the Validate signature to accept an optional handler.

Reply:
I'm adding a private field `workloadType` to `AdvancedCount` to help us narrow down which fields we should be checking

Reply:
OK ! @Lou1415926 and @iamhopaul123 can you both take a look at this commit for the changes? https://github.com/aws/copilot-cli/pull/2827/commits/87588b67ae6a0b7b71e23db2b213b7b51814325f 🥺